modules_core/com.smf.securewebservices/src/com/smf/securewebservices/utils/SecureDataToJson.java

```diff
@@ -68,6 +76,12 @@ public JSONObject toJsonObject(BaseOBObject bob, DataResolvingMode dataResolving
 			final boolean isDerivedReadable = OBContext.getOBContext().getEntityAccessChecker()
 					.isDerivedReadable(bob.getEntity());
 
+			// validate if this entity is readable.
+			if (OBContext.getOBContext().isInAdministratorMode()) {
+				OBContext.restorePreviousMode();
```

Comment:
It would be better to restore the adminMode to the exact mode setted before calling it


if (OBContext.getOBContext().isInAdministratorMode()) {
 var orgClientAccessCheck = OBContext.getOBContext().doOrgClientAccessCheck();
 OBContext.restorePreviousMode();
 OBContext.getOBContext().getEntityAccessChecker().checkReadable(bob.getEntity());
 OBContext.setAdminMode(orgClientAccessCheck);
} else {
  OBContext.getOBContext().getEntityAccessChecker().checkReadable(bob.getEntity());
}

Reply:
tested and working